keymanapp/keyman
# fix(oem/fv/android): Consolidate dictionary and keyboard settings

Follow-on to #6282
For the FV Android app, this consolidates the model picker with the Keyboard settings menu so the user has less clicks.

Keyman Engine API change:
* make deleteLexicalModel() public

## Updated screenshot
![new keyboard settings](https://user-images.githubusercontent.com/7358010/158092845-f9acf2af-a785-422b-bd31-b7c8b0687cdc.png)

Tagging @madebybridget

## User Testing

* **TEST_MENU**
1. Load the PR build of the FirstVoices Android app and launch
2. On the main setup page, click "Select keyboards" --> BC Coast --> SENCOTEN
3. On the "SENCOTEN Settings" page, enable the keyboard
4. Wait for the associated dictionary to download and install (this could take a minute)
5. Verify the "SENCOTEN Settings" menu updates to show
    * Keyboard is enabled
    * Predictions is enabled
    * Corrections is enabled
    * SENCOTEN dictionary is installed (left blue checkmark)
6. Click on "SENCOTEN dictionary"
7. Verify the "SENCOTEN dictionary" info page appears
8. Click back to return to "SENCOTEN Settings"
9. On the device, disable network access
10. On the "SENCOTEN Settings" menu, click on "Check for dictionaries online"
11. Verify the app notifies it cannot connect to Keyman server
12. On the device, re-enable network access
13. On the "SENCOTEN Settings" menu, click on "Check for dictionaries online"
14. Verify the app queries and downloads the dictionary again.
